python/ray/data/_internal/planner/plan_udf_map_op.py

```diff
@@ -318,13 +318,18 @@ def transform_fn(
         # generators, and in the main event loop, yield them from
         # the queue as they become available.
         output_batch_queue = queue.Queue()
+        exception_queue = queue.Queue()
 
         async def process_batch(batch: DataBatch):
-            output_batch_iterator = await fn(batch)
-            # As soon as results become available from the async generator,
-            # put them into the result queue so they can be yielded.
-            async for output_batch in output_batch_iterator:
-                output_batch_queue.put(output_batch)
+            try:
+                output_batch_iterator = await fn(batch)
+                # As soon as results become available from the async generator,
+                # put them into the result queue so they can be yielded.
+                async for output_batch in output_batch_iterator:
+                    output_batch_queue.put(output_batch)
+            except Exception as e:
+                exception_queue.put(e)
+                output_batch_queue.put(None)  # Signal to stop processing
```

Comment:
nit: could we use a constant/more specific sentinel value instead of generic `None` in this case?

Reply:
I just change it to dict